### DISCUSSION TITLE
Silly question...

### DISCUSSION THREAD

**Discussion by Rickwright12** [Budibase/budibase#2808] [Help] [answered]
My apologies for having to ask this, but am new to docker and new to Budibase, but it looks like the ideal tool I need for a project I am working on.
I keep getting an error when I go to ./budi hosting --start
"The WATCHTOWER_PORT variable is not set. Defaulting to a blank string.
The Compose file './docker-compose.yaml' is invalid because:
services.watchtower-service.ports contains an invalid type, it should be a number, or an object"

I think it is a docker error, but am unsure.
Has anyone else had this?

regards
Rick

**Comment by tobehn** [comment, marked as answer]
@Rickwright12, did you use docker-compose?
Ports and other settings are declared in hosting.properties and envoy.yaml files.

See the documentation here:
https://docs.budibase.com/self-hosting/hosting-methods/docker-compose

After setting up docker, docker-compose and downloading [docker-compose.yaml](https://raw.githubusercontent.com/Budibase/budibase/master/hosting/docker-compose.yaml), [hosting.properties](https://raw.githubusercontent.com/Budibase/budibase/master/hosting/hosting.properties) and [envoy.yaml](https://raw.githubusercontent.com/Budibase/budibase/master/hosting/envoy.yaml) into one directory you should be able to fire it up with

`docker-compose --env-file hosting.properties up`
remember setting `-d` to have it running as daemon

**Reply by Rickwright12** [reply]
Many thanks indeed.  For some reason I was focusing on Budibase CLI Setup and having a very vacant afternoon!
Many thanks for your help, up and running now
